### ISSUE TITLE
`plt.show()` ignored

### ISSUE THREAD

**Issue by OverLordGoldDragon** [sphinx-gallery/sphinx-gallery#1025]
```python
plt.plot([1, 2]); plt.show()
plt.plot([2, 1]); plt.show()
```
generates only one figure. Explicitly creating a `figure` works, but requires reworking existing code. Maybe it's something with matplotlib backends (which I'm not familiar with), but `plt.ion()` doesn't fix this - in Jupyter and Spyder I have the `inline` backend, with

```
plt.get_backend() == 'module://matplotlib_inline.backend_inline'
```
but that doesn't seem settable via `matplotlib.use()`. Can my desired behavior be achieved (create new figs after `plt.show()`)?

**Comment by lucyleeow** [comment]
I tried the above code and it seem that matplotlib plots both lines on the same plot? There is only one figure generated?

![sphx_glr_plot_ex1_001](https://user-images.githubusercontent.com/23182829/192924384-5ed1556a-41c3-4686-aea4-3c6d4ac83424.png)

**Comment by OverLordGoldDragon** [comment]
Yes so your configuration agrees with what `sphinx-gallery` uses, but it's not standard interactive Python behavior; `plt.show()` should show figures up to current point, and subsequent `plt.` calls generate a new figure.

**Comment by GaelVaroquaux** [comment]
I believe that this is quite similar to how jupyter works, and we are mimicking the behavior of jupyter.

**Comment by lucyleeow** [comment]
@larsoner I didn't realise this. Is it because of how we execute the code blocks?

**Comment by OverLordGoldDragon** [comment]
@GaelVaroquaux I don't think so, I've never seen this behavior as the default in Jupyter. My code produces two plots in the [official intro](https://jupyter.org/try-jupyter/retro/notebooks/?path=notebooks/Intro.ipynb) and [Google Colab](https://colab.research.google.com/).

@larsoner Since SG strives to make ipynb out of py, this strikes me as significant deviation, as expected behavior isn't reproduced by SG. For my visuals-heavy codebase which assumes interactive behavior, for example, I'll have to rewrite a bunch of code to accomodate SG. If within reason, I think SG should make this a configurable. Either way, is there a to override this behavior as of now?

**Comment by larsoner** [comment]
Anything you want to separate split into a new code block for example with
```
# %%
```
If you put this between your two show calls it should produce two figures

**Comment by OverLordGoldDragon** [comment]
I'm aware but this messes with the presentation format.
